### PR TITLE
[RFC] Question of Tooling Consistency Check Between Agent<>AgentExecutor

### PR DESCRIPTION
**Pull request by skcoirz**
This is a follow up of https://github.com/hwchase17/langchain/pull/3840

Question: Why do we allow `None` `allowed_tools` in `Agent` when there are tools provided in `AgentExecutor`?

AgentExecutor is able to fetch the `allowed_tools` from `Agent` directly. It doesn’t look like we need to provide `tools` again at `AgentExecutor` entry functions when `Agent` is already provided as a parameter.

I’m new to the logics. I could be wrong. Would like to get some help here. Thank you so much.

### CODE REVIEW

File: langchain/agents/agent.py [hunk 1]
```diff
@@ -629,12 +629,15 @@ def validate_tools(cls, values: Dict) -> Dict:
         agent = values["agent"]
         tools = values["tools"]
         allowed_tools = agent.get_allowed_tools()
-        if allowed_tools is not None:
-            if set(allowed_tools) != set([tool.name for tool in tools]):
-                raise ValueError(
-                    f"Allowed tools ({allowed_tools}) different than "
-                    f"provided tools ({[tool.name for tool in tools]})"
-                )
+        if allowed_tools is None and len(tools) == 0:
```
Comment (dev2049) [inline langchain/agents/agent.py:632]: don't think this is needed

**Comment by skcoirz** [comment]
Thank you for the review, dev2049. I have some questions here as described
in the PR summary. Can I ask why you think it’s not needed?

**Comment by hwchase17** [comment]
i also dont think this is neccessary. allowed_tools can be None for a myriad of reasons. if its None we should just ignore

**Comment by skcoirz** [comment]
@hwchase17 , @dev2049 , I’m okay with closing this PR, but when you have time, can you explain a bit when `allowed_tools` should be `None`? Ideally with a few real use cases? I have checked the logics of Agent and AgentExecutor, but I couldn’t understand why we allow it to be `None`. Would appreciate your context sharing. Thank you.

**Comment by skcoirz** [comment]
not hearing back from the core team. close for now.

**Pull request closed without merging**